docs/enhancements/aws-private-link.md

```diff
@@ -0,0 +1,553 @@
+with the same ServerCertificate `*.<hosted zone domain>`.
+
+We add the APIServer and Secret object with correct configuration during the
+manifests stage of openshift-install and the cluster apiserver will setup the
+listeners to respond correctly.
+
+### Approving VPC Endpoints
+
+Endpoints services can require that any endpoints created to the service be
+automatically accepted or manually approved. Now since, these endpoint services
+are only visible (or accessible) to users defined by the AllowedPrincipals list for
+the service if only very specific user/role/account is allowed to create
+endpoints for these services, using automatic approval is a safe option.
+
+```shell=bash
+$ aws ec2 create-vpc-endpoint-service-configuration --no-acceptance-required
+```
+```shell=bash
+$ aws ec2 modify-vpc-endpoint-service-permissions \
+--add-allowed-principals '["arn:aws:iam::123456789012:root"]' 
+```
+
+Effectively only allow users in 123456789012 account to create endpoints to the service.
+NOTE: This can be restricted to a specific user/role in the account too.
+
+### Manual walk-through
+
+Here is a manual walk through of all the steps that need to be taken today
+to setup the required environment.
+
+So let's assume we have a public Route53 HostedZone `hub01.osdprivate.io`. And
+the computes DNS name for the endpoint service is `api.hash123.hub01osdprivate.io`.
+
+Create the manifests for install.
+```shell=bash
+$ openshift-install --dir install_dir01 create manifests
+```
+
+Add the APIServer and Secret object for setting up apiserver SNI.
+```shell=bash
+$ cat > install_dir01/manifests/apiserver.yaml << EOF
+apiVersion: config.openshift.io/v1
+kind: APIServer
+metadata:
+  name: cluster
+spec:
+  servingCerts:
+    namedCertificates:
+    - names:
+      - api.hash123.hub01osdprivate.io
+      servingCertificate:
+       name: apiserver-private-link-ingress
+EOF
+$ cat > install_dir01/manifests/secret-apiserver-private-link-ingress.yaml << EOF
+apiVersion: v1
+kind: Secret
+metadata:
+  namespace: openshift-config
+  name: apiserver-private-link-ingress
+tls.key: <the serving certificate key here>
+tls.crt: <the serving certificate here>
+EOF
+```
+
+Create infrastructure resources only.
+```shell=bash
+$ openshift-install --dir install_dir01 create cluster --no-wait-only-infra
+```
+
+At this stage the installer has created the NLB for the apiserver and the
+installation is already in progress in the bootstrap ec2instance. We can use
+the installer tags to figure out the apiserver's internal NLB for the cluster.
+
+Create an Endpoint service in AWS account of the cluster.
+```shell=bash
+$ AWS_PROFILE=customer aws ec2 create-vpc-endpoint-service-configuration \
+--no-acceptance-required \
+--private-dns-name api.hash123.hub01osdprivate.io \
+--network-load-balancer-arns <nlb arn here>
+```
+```shell=bash
+$ aws ec2 modify-vpc-endpoint-service-permissions \
+--add-allowed-principals '["arn:aws:iam::123456789012:root"]' 
```

Comment:
https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_ModifyVpcEndpointServicePermissions.html

The VPC Endpoint Service needs to define which accounts/users/roles can create an VPC Endpoint to it..  and this makes sure the either
- entire Hub AWS Account can create the VPC Endpoint or
- the specific user whose credentials are being used to create VPC Endpoint be allowed to do so.
since for us we have this user information when we create the VPC Endpoint Service, it makes sense to only allow it that permission.